src/test/java/com/thealgorithms/bitmanipulation/HighestSetBitTest.java

```diff
@@ -0,0 +1,20 @@
+package com.thealgorithms.bitmanipulation;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test case for Highest Set Bit
+ * @author Bama Charan Chhandogi (https://github.com/BamaCharanChhandogi)
+ */
+
+class HighestSetBitTest {
+
+    @Test
+    void testHighestSetBit() {
```

Comment:
What about negative numbers?

Reply:
Yes. position of the highest set bit of negative number will be the same.
I have added test case

Reply:
Negative numbers have sign bit set, so I'd say it should return 31 for them

Reply:
Alternatively, you can change input to be unsigned to avoid negative numbers

Reply:
> Alternatively, you can change input to be unsigned to avoid negative numbers

Does Java have `unsigned`?

I would suggest to throw `IllegalArgumentException` for negative inputs.

Reply:
Let's go with `IllegalArgumentException` for negative inputs